src/libsync/discovery.cpp

```diff
@@ -177,9 +180,15 @@ void ProcessDirectoryJob::process()
         // local stat function.
         // Recall file shall not be ignored (#4420)
         bool isHidden = e.localEntry.isHidden || (f.first[0] == '.' && f.first != QLatin1String(".sys.admin#recall#"));
+#ifdef Q_OS_WIN
+        // exclude ".lnk" files as they are not essential, but, causing troubles when enabling the VFS due to QFileInfo::isDir() and other methods are freezing, which causes the ".lnk" files to start hydrating and freezing the app eventually.
+        const bool isServerEntrySymLink = !e.localEntry.isValid() && e.serverEntry.isValid() && !e.serverEntry.isDirectory && FileSystem::isLnkFile(e.serverEntry.name);
+#else
+        const bool isServerEntrySymLink = false;
+#endif
```

Comment:
I wonder if it is a good idea to move the preprocessor conditionals into the `FileSystem::isLnkFile()` method? That would simplify the logic a bit.

Reply:
@FlexW 
In this case, I wouldn't do that, cause, "isLnkFile" can be used on other platforms too.